src/averageAges.js

```diff
@@ -15,11 +15,13 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => (!century
+    || Math.ceil(person.died / 100) === century)
+    && person.sex === 'm');
+  const averageAge = men.reduce((acc, person) =>
+    acc + (person.died - person.born), 0);
+
+  return averageAge / men.length;
```

Comment:
you can create separate function to calculate the average age based on array param and reuse it here and in `calculateWomenAverageAge`